disco/disco.go

```diff
@@ -204,7 +214,7 @@ func (d *Disco) discover(hostname svchost.Hostname) (*Host, error) {
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to request discovery document: %v", err)
+		return nil, &ErrServiceDiscoveryNetworkRequest{err}
```

Comment:
You can return a value type for the `error` interface

```suggestion
		return nil, ErrServiceDiscoveryNetworkRequest{err}
```